prefect_dbt/cloud/jobs.py

```diff
@@ -87,10 +143,174 @@ def trigger_job_run_flow():
+        ```
+    """  # noqa
+    try:
+        async with dbt_cloud_credentials.get_administrative_client() as client:
+            response = await client.get_run(run_id=run_id)
+    except HTTPStatusError as ex:
+        raise DbtCloudGetRunFailed(extract_user_message(ex)) from ex
+    return response.json()["data"]
+
+
+@flow
+async def trigger_dbt_cloud_job_run_and_wait_for_completion(
+    dbt_cloud_credentials: DbtCloudCredentials,
+    job_id: int,
+    trigger_job_run_options: Optional[TriggerJobRunOptions] = None,
+    max_wait_seconds: Optional[int] = None,
+    poll_frequency_seconds: int = 10,
+) -> Dict:
+    """
+    Flow that triggers a job run and waits for the triggered run to complete.
+
+    Args:
+        dbt_cloud_credentials: Credentials for authenticating with dbt Cloud.
+        job_id: The ID of the job to trigger.
+        trigger_job_run_options: An optional TriggerJobRunOptions instance to
+            specify overrides for the triggered job run.
+        max_wait_seconds: Maximum number of seconds to wait for job to complete
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Returns:
+        The run data returned by the dbt Cloud administrative API.
+
+    Example:
+        Trigger a dbt Cloud job and wait for completion as a stand alone flow:
+        ```python
+        import asyncio
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import trigger_dbt_cloud_job_run_and_wait_for_completion
+
+        asyncio.run(
+            trigger_dbt_cloud_job_run_and_wait_for_completion(
+                dbt_cloud_credentials=DbtCloudCredentials(
+                    api_key="my_api_key",
+                    account_id=123456789
+                ),
+                job_id=1
+            )
+        )
+        ```
+
+        Trigger a dbt Cloud job and wait for completion as a sub-flow:
+        ```python
+        from prefect import flow
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import trigger_dbt_cloud_job_run_and_wait_for_completion
+
+        @flow
+        def my_flow():
+            ...
+            run_result = trigger_dbt_cloud_job_run_and_wait_for_completion(
+                dbt_cloud_credentials=DbtCloudCredentials(
+                    api_key="my_api_key",
+                    account_id=123456789
+                ),
+                job_id=1
+            )
+            ...
+
+        my_flow()
+        ```
+
+    """  # noqa
+    trigger_job_run_state = await trigger_dbt_cloud_job_run(
+        dbt_cloud_credentials=dbt_cloud_credentials,
+        job_id=job_id,
+        options=trigger_job_run_options,
+    )
+    triggered_run_data = await trigger_job_run_state.result()
+    run_id = triggered_run_data["id"]
+
+    seconds_waited_for_run_completion = 0
+    while not max_wait_seconds or seconds_waited_for_run_completion <= max_wait_seconds:
```

Comment:
I opted to change `max_wait_seconds` to `int` with a default of `900` to simplify the code and because we probably don't want the flow to wait forever by default.